plugins/analytics-module-ga/src/apis/implementations/AnalyticsApi/GoogleAnalytics.test.ts

```diff
@@ -152,6 +153,101 @@ describe('GoogleAnalytics', () => {
       });
     });
 
+    it('captures virtual pageviews instead of search events', () => {
+      const config = new ConfigReader({
+        app: {
+          analytics: {
+            ga: {
+              trackingId,
+              testMode: true,
+              virtualSearchPageView: { mode: 'only' },
+            },
+          },
+        },
+      });
+      const api = GoogleAnalytics.fromConfig(config);
+      api.captureEvent({
+        action: 'search',
+        subject: 'test search',
+        context,
+      });
+
+      const [command, data] = ReactGA.testModeAPI.calls[1];
+      expect(command).toBe('send');
+      expect(data).toMatchObject({
+        hitType: 'pageview',
+        page: '/search?query=test+search',
+      });
+      expect(ReactGA.testModeAPI.calls).toHaveLength(2);
+    });
+
+    it('captures virtual pageviews alongside search events', () => {
+      const config = new ConfigReader({
+        app: {
+          analytics: {
+            ga: {
+              trackingId,
+              testMode: true,
+              virtualSearchPageView: { mode: 'both' },
+            },
+          },
+        },
+      });
+      const api = GoogleAnalytics.fromConfig(config);
+      api.captureEvent({
+        action: 'search',
+        subject: 'test search',
+        context,
+      });
+
+      const [pageviewCommand, pageViewData] = ReactGA.testModeAPI.calls[1];
+      expect(pageviewCommand).toBe('send');
+      expect(pageViewData).toMatchObject({
+        hitType: 'pageview',
+        page: '/search?query=test+search',
+      });
+      const [searchCommand, searchData] = ReactGA.testModeAPI.calls[2];
+      expect(searchCommand).toBe('send');
+      expect(searchData).toMatchObject({
+        hitType: 'event',
+        eventCategory: context.extension,
+        eventAction: 'search',
+        eventLabel: 'test search',
+      });
+    });
+
+    it('captures virtual pageviews on custom route with custom search query and custom category', () => {
+      const config = new ConfigReader({
+        app: {
+          analytics: {
+            ga: {
+              trackingId,
+              testMode: true,
+              virtualSearchPageView: {
+                mode: 'only',
+                mountPath: '/custom',
+                searchQuery: 'term',
+                categoryQuery: 'sc',
+              },
+            },
+          },
+        },
+      });
+      const api = GoogleAnalytics.fromConfig(config);
+      api.captureEvent({
+        action: 'search',
+        subject: 'test search',
+        context,
+      });
+
+      const [command, data] = ReactGA.testModeAPI.calls[1];
+      expect(command).toBe('send');
+      expect(data).toMatchObject({
+        hitType: 'pageview',
+        page: '/custom?term=test+search&sc=test+category',
```

Comment:
Hmmmm....  Now I'm second guessing myself!  Do you know if GA treats `test+search`, `test search`, and `test%20search` the same when captured as the page on a pageview?

Reply:
Quick test seems to indicate differentiation, at least in the real-time view, but I don't know if/how it trickles down into site search (I don't have an easy way to test that just now)

![Screenshot 2022-11-30 at 12 00 05](https://user-images.githubusercontent.com/3496491/204780403-b76d64e9-887d-426f-816d-35762fce2602.png)

Reply:
![image](https://user-images.githubusercontent.com/3626328/204783973-64235eb4-014e-4333-bd9b-84066ca47fae.png)
In Site Search it does get parsed correctly, I've checked

Reply:
But we can also switch back to the old version without `URLSearchParams` if you'd like

Reply:
Nah, all good!  Just wanted to be sure.